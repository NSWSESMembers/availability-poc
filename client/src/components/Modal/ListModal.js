import React, { Component } from 'react';

import PropTypes from 'prop-types';
import { View, Modal, TouchableOpacity, TouchableWithoutFeedback, Text, ScrollView } from 'react-native';

import styles from './styles';

let componentIndex = 0;

class ListModal extends Component {
  renderOption(option, isLastItem) {
    return (
      <TouchableOpacity key={option.key} onPress={() => this.props.onChange(option)}>
        <View style={[styles.optionStyle, isLastItem &&
            { borderBottomWidth: 0 }]}
        >
          <Text style={styles.optionTextStyle}>{option.label}</Text>
        </View>
      </TouchableOpacity>);
  }

  renderOptionList() {
    const options = this.props.data.map(
      (item, index) => this.renderOption(
        item, index === this.props.data.length - 1,
      ),
    );

    componentIndex += 1;

    return (
      <TouchableWithoutFeedback
        key={`modalSelector${componentIndex}`}
        onPress={() => this.props.closeModal}
      >
        <View style={styles.overlayStyle}>
          <View style={styles.headerContainer}>
            <TouchableOpacity>
              <View style={styles.headerStyle}>
                <Text style={styles.headerTextStyle}>{this.props.title}</Text>
              </View>
            </TouchableOpacity>
          </View>
          <View style={styles.optionContainer}>
            <ScrollView>
              <View style={{ paddingHorizontal: 10 }}>
                {options}
              </View>
            </ScrollView>
          </View>
          <View style={styles.cancelContainer}>
            <TouchableOpacity onPress={this.props.backModal}>
              <View style={styles.cancelStyle}>
                <Text style={styles.cancelTextStyle}>Back</Text>
              </View>
            </TouchableOpacity>
          </View>
        </View>
      </TouchableWithoutFeedback>);
  }


  render() {
    return (
      <View>
        <Modal
          transparent
          visible={this.props.visible}
          onRequestClose={this.props.closeModal}
          animationType="fade"
        >
          {this.renderOptionList()}
        </Modal>
      </View>
    );
  }
}
ListModal.propTypes = {
  title: PropTypes.string.isRequired,
  data: PropTypes.array, // eslint-disable-line react/forbid-prop-types
  closeModal: PropTypes.func.isRequired,
  backModal: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  visible: PropTypes.bool.isRequired,
};

export default ListModal;
