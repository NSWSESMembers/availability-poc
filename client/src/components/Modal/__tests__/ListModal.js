import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

import ListModal from '../ListModal';

const noop = () => {};

test('renders correctly', () => {
  const tree1 = renderer.create(
    <ListModal
      title="test"
      data={[<Text>test</Text>]}
      closeModal={noop}
      backModal={noop}
      onChange={noop}
      visible
    />,
  ).toJSON();
  expect(tree1).toMatchSnapshot();
});
