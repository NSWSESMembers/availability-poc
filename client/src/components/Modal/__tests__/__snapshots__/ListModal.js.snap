// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={true}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.7)",
          "flex": 1,
          "justifyContent": "center",
          "padding": "5%",
        }
      }
      testID={undefined}
    >
      <View
        style={
          Object {
            "alignSelf": "stretch",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(255,255,255,0.8)",
                "borderRadius": 5,
                "padding": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#333",
                  "fontSize": 16,
                  "textAlign": "center",
                }
              }
            >
              test
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.8)",
            "borderRadius": 5,
            "flexShrink": 1,
            "marginBottom": 8,
            "marginTop": 8,
            "padding": 8,
          }
        }
      >
        <RCTScrollView>
          <View>
            <View
              style={
                Object {
                  "paddingHorizontal": 10,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderBottomColor": "#ccc",
                        "borderBottomWidth": 1,
                        "padding": 8,
                      },
                      Object {
                        "borderBottomWidth": 0,
                      },
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "rgba(0,118,255,0.9)",
                        "fontSize": 16,
                        "textAlign": "center",
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Object {
            "alignSelf": "stretch",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(255,255,255,0.8)",
                "borderRadius": 5,
                "padding": 8,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#333",
                  "fontSize": 16,
                  "textAlign": "center",
                }
              }
            >
              Back
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
